crates/polars-utils/src/pl_str.rs

```diff
@@ -0,0 +1,124 @@
+use compact_str::CompactString;
+
+#[macro_export]
+macro_rules! format_pl_smallstr {
+    ($($arg:tt)*) => {{
+        use std::fmt::Write;
+
+        let mut string = PlSmallStr::const_default();
+        write!(string, $($arg)*).unwrap();
+        string
+    }}
+}
+
+/// String type that interns small strings and has O(1) clone.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
```

Comment:
Instead of deriving `Debug` can you implement it manually so it prints nicely as a normal `String` would? Otherwise it adds just a bunch of noise that no one really cares about.